registry/coder/modules/amazon-q/scripts/install.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+# Install script for amazon-q module
+
+set -o errexit
+set -o pipefail
+
+command_exists() {
+  command -v "$1" >/dev/null 2>&1
+}
+
+# Inputs
+ARG_INSTALL=${ARG_INSTALL:-true}
+ARG_VERSION=${ARG_VERSION:-latest}
+ARG_AUTH_TARBALL=${ARG_AUTH_TARBALL:-}
+ARG_AGENT_CONFIG=${ARG_AGENT_CONFIG:-}
+ARG_MODULE_DIR_NAME=${ARG_MODULE_DIR_NAME:-.aws/.amazonq}
+ARG_CODER_MCP_APP_STATUS_SLUG=${ARG_CODER_MCP_APP_STATUS_SLUG:-}
+ARG_PRE_INSTALL_SCRIPT=${ARG_PRE_INSTALL_SCRIPT:-}
+ARG_POST_INSTALL_SCRIPT=${ARG_POST_INSTALL_SCRIPT:-}
+
+mkdir -p "$HOME/$ARG_MODULE_DIR_NAME"
+
+# Decode base64 inputs
+ARG_AGENT_CONFIG_DECODED=""
+if [ -n "$ARG_AGENT_CONFIG" ]; then
+  ARG_AGENT_CONFIG_DECODED=$(echo -n "$ARG_AGENT_CONFIG" | base64 -d)
+fi
+
+echo "--------------------------------"
+echo "install: $ARG_INSTALL"
+echo "version: $ARG_VERSION"
+echo "coder_mcp_app_status_slug: $ARG_CODER_MCP_APP_STATUS_SLUG"
+echo "module_dir_name: $ARG_MODULE_DIR_NAME"
+echo "auth_tarball_provided: $([ -n "$ARG_AUTH_TARBALL" ] && echo "yes" || echo "no")"
+echo "pre_install_script_provided: $([ -n "$ARG_PRE_INSTALL_SCRIPT" ] && echo "yes" || echo "no")"
+echo "post_install_script_provided: $([ -n "$ARG_POST_INSTALL_SCRIPT" ] && echo "yes" || echo "no")"
+echo "--------------------------------"
+
+# Execute pre-install script if provided
+function pre_install() {
+  if [ -n "$ARG_PRE_INSTALL_SCRIPT" ]; then
+    echo "Executing pre-install script..."
+    # Decode base64 encoded script and execute it
+    echo "$ARG_PRE_INSTALL_SCRIPT" | base64 -d > /tmp/pre_install.sh
+    chmod +x /tmp/pre_install.sh
+    /tmp/pre_install.sh
+    rm -f /tmp/pre_install.sh
+    echo "Pre-install script completed successfully."
+  else
+    echo "No pre-install script provided, skipping..."
+  fi
+}
+
+# Install Amazon Q if requested
+function install_amazon_q() {
+  if [ "$ARG_INSTALL" = "true" ]; then
+    echo "Installing Amazon Q..."
+    PREV_DIR="$PWD"
+    TMP_DIR="$(mktemp -d)"
+    cd "$TMP_DIR"
+
+    ARCH="$(uname -m)"
+    case "$ARCH" in
+    "x86_64")
+      Q_URL="https://desktop-release.q.us-east-1.amazonaws.com/${ARG_VERSION}/q-x86_64-linux.zip"
```

Comment:
Just thinking about enterprise use cases where direct internet access is unlikely to be allowed, we should consider the introduction of a new var for specifying the Q install URL Prefix.

ie, var.q_install_url with a default value of "https://desktop-release.q.us-east-1.amazonaws.com" so that users can control this. The same naming format, etc that Amazon us for versioning and filenames still apply however.